ee/desktop/desktop.go

```diff
@@ -0,0 +1,18 @@
+package desktop
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+)
+
+func DesktopSocketPath(pid int) string {
```

Comment:
This is a bit of a can of worms, and I'm not sure what correct is. To talk aloud....

On windows, we have to be in `\\.pipe`, okay. We'll just need to pick a good name and roll with it.  Maybe add something to detect collisions. Not sure.

On not-windows, it feels weird to use TempDir. Nothing really cleans it up, just feels wrong. I'm not sure what feels right.

In one world, launcher would create the pipe somewhere, chown/chmod it to the user, and then hand it to desktop for startup. Other worlds might be creating a user writable run dir.

Reply:
what about `$HOME/.kolide/` and put the socket there? launcher could create the folder and give user privs.

Reply:
If we stored user preferences we could think about it. But it feels like bad form for this.